src/EnergyPlus/OutputReports.cc

```diff
@@ -348,85 +322,87 @@ static void WriteDXFCommon(EnergyPlusData &state, InputOutputFile &of, const std
     Real64 minx = 99999.0;
     Real64 miny = 99999.0;
     for (int surf : state.dataSurface->AllSurfaceListReportOrder) {
-        if (state.dataSurface->Surface(surf).Class == SurfaceClass::IntMass) continue;
-        for (int vert = 1; vert <= state.dataSurface->Surface(surf).Sides; ++vert) {
-            minx = min(minx, state.dataSurface->Surface(surf).Vertex(vert).x);
-            miny = min(miny, state.dataSurface->Surface(surf).Vertex(vert).y);
+        auto &thisSurface = state.dataSurface->Surface(surf);
+        if (thisSurface.Class == DataSurfaces::SurfaceClass::IntMass) continue;
+        for (int vert = 1; vert <= thisSurface.Sides; ++vert) {
+            minx = min(minx, thisSurface.Vertex(vert).x);
+            miny = min(miny, thisSurface.Vertex(vert).y);
         }
     }
 
-    for (int vert = 1; vert <= 4; ++vert) {
-        StemX(vert) += minx;
-        StemY(vert) += miny;
-        Head1X(vert) += minx;
-        Head1Y(vert) += miny;
-        Head2X(vert) += minx;
-        Head2Y(vert) += miny;
-        NSide1X(vert) += minx;
-        NSide1Y(vert) += miny;
-        NSide2X(vert) += minx;
-        NSide2Y(vert) += miny;
-        NSide3X(vert) += minx;
-        NSide3Y(vert) += miny;
+    for (int vert = 0; vert < lenArr; ++vert) {
+        StemX[vert] += minx;
+        StemY[vert] += miny;
+        Head1X[vert] += minx;
+        Head1Y[vert] += miny;
+        Head2X[vert] += minx;
+        Head2Y[vert] += miny;
+        NSide1X[vert] += minx;
+        NSide1Y[vert] += miny;
+        NSide2X[vert] += minx;
+        NSide2Y[vert] += miny;
+        NSide3X[vert] += minx;
+        NSide3Y[vert] += miny;
```

Comment:
I wonder the best way to do this from a performance standpoint.  I am _not_ suggesting we make a change to this loop here, but it's a good opportunity for chatting about it.  Is it better to loop by index and go array-by-array?  Seems like it would be better to do a tight loop on each array, but maybe the loop set-up could make that less valuable.  I worked up an example to convince myself:

```
#include <algorithm>
#include <chrono>
#include <iostream>
#include <vector>

constexpr unsigned int lenArr = 1000;
std::vector<float> StemX;
std::vector<float> StemY;
std::vector<float> Head1X;
std::vector<float> Head1Y;
std::vector<float> Head2X;
std::vector<float> Head2Y;
std::vector<float> NSide1X;
std::vector<float> NSide1Y;
std::vector<float> NSide2X;
std::vector<float> NSide2Y;
std::vector<float> NSide3X;
std::vector<float> NSide3Y;

void initVectors() {
    StemX.resize(lenArr);
    StemY.resize(lenArr);
    Head1X.resize(lenArr);
    Head1Y.resize(lenArr);
    Head2X.resize(lenArr);
    Head2Y.resize(lenArr);
    NSide1X.resize(lenArr);
    NSide1Y.resize(lenArr);
    NSide2X.resize(lenArr);
    NSide2Y.resize(lenArr);
    NSide3X.resize(lenArr);
    NSide3Y.resize(lenArr);
}

void multipleVectorsAtOnce(float const minx, float const miny) {
    for (int vert = 0; vert < lenArr; ++vert) {
        StemX[vert] += minx;
        StemY[vert] += miny;
        Head1X[vert] += minx;
        Head1Y[vert] += miny;
        Head2X[vert] += minx;
        Head2Y[vert] += miny;
        NSide1X[vert] += minx;
        NSide1Y[vert] += miny;
        NSide2X[vert] += minx;
        NSide2Y[vert] += miny;
        NSide3X[vert] += minx;
        NSide3Y[vert] += miny;
    }
}

void vectorsSeparately(float const minx, float const miny) {
    for (int vert = 0; vert < lenArr; ++vert) { StemX[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { StemY[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { Head1X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { Head1Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { Head2X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { Head2Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide1X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide1Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide2X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide2Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide3X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide3Y[vert] += miny; }
}

void usingTransform(float const minx, float const miny) {
    std::transform(StemX.begin(), StemX.end(), StemX.begin(), [minx](float x){return x + minx;});
    std::transform(StemX.begin(), StemX.end(), StemX.begin(), [miny](float x){return x + miny;});
    std::transform(Head1X.begin(), Head1X.end(), Head1X.begin(), [minx](float x){return x + minx;});
    std::transform(Head1Y.begin(), Head1Y.end(), Head1Y.begin(), [miny](float x){return x + miny;});
    std::transform(Head2X.begin(), Head2X.end(), Head2X.begin(), [minx](float x){return x + minx;});
    std::transform(Head2Y.begin(), Head2Y.end(), Head2Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide1X.begin(), NSide1X.end(), NSide1X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide1Y.begin(), NSide1Y.end(), NSide1Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide2X.begin(), NSide2X.end(), NSide2X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide2Y.begin(), NSide2Y.end(), NSide2Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide3X.begin(), NSide3X.end(), NSide3X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide3Y.begin(), NSide3Y.end(), NSide3Y.begin(), [miny](float x){return x + miny;});
}

int main() {
    constexpr unsigned int timesToRepeat = 10000;
    std::chrono::time_point<std::chrono::system_clock> start, end;
    
    std::cout << "Initializing Vectors... ";
    initVectors();
    std::cout << "Done" << std::endl;

    std::cout << "Aggregating Vectors 1... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        multipleVectorsAtOnce(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration2 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration2 << "ms" << std::endl;

    std::cout << "Aggregating Vectors 2... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        vectorsSeparately(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration3 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration3 << "ms" << std::endl;

    std::cout << "Aggregating Vectors 3... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        usingTransform(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration4 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration4 << "ms" << std::endl;

    return 0;
}
```

When built and run in release mode, it gives me this:

```
Initializing Vectors... Done
Aggregating Vectors 1... Done in 25ms
Aggregating Vectors 2... Done in 10ms
Aggregating Vectors 3... Done in 11ms
```

So it looks like the current way is 2.5x slower than either the individual loops or the transform method.

**OF COURSE** the impact of this depends on the context, the size of the vectors and how often we do this, I just nerd sniped myself and thought I'd share what I found and see if it led to fun discussion.

Reply:
Great catch, I didn't even grok that you had converted them, and that the size was fixed and so small.  I modified the code for std::array of size 4, bumped up the number of times it ran, and got new results:

```
#include <array>
#include <algorithm>
#include <chrono>
#include <iostream>

constexpr unsigned int lenArr = 4;
std::array<float, lenArr> StemX;
std::array<float, lenArr> StemY;
std::array<float, lenArr> Head1X;
std::array<float, lenArr> Head1Y;
std::array<float, lenArr> Head2X;
std::array<float, lenArr> Head2Y;
std::array<float, lenArr> NSide1X;
std::array<float, lenArr> NSide1Y;
std::array<float, lenArr> NSide2X;
std::array<float, lenArr> NSide2Y;
std::array<float, lenArr> NSide3X;
std::array<float, lenArr> NSide3Y;

void multipleVectorsAtOnce(float const minx, float const miny) {
    for (int vert = 0; vert < lenArr; ++vert) {
        StemX[vert] += minx;
        StemY[vert] += miny;
        Head1X[vert] += minx;
        Head1Y[vert] += miny;
        Head2X[vert] += minx;
        Head2Y[vert] += miny;
        NSide1X[vert] += minx;
        NSide1Y[vert] += miny;
        NSide2X[vert] += minx;
        NSide2Y[vert] += miny;
        NSide3X[vert] += minx;
        NSide3Y[vert] += miny;
    }
}

void vectorsSeparately(float const minx, float const miny) {
    for (int vert = 0; vert < lenArr; ++vert) { StemX[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { StemY[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { Head1X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { Head1Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { Head2X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { Head2Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide1X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide1Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide2X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide2Y[vert] += miny; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide3X[vert] += minx; }
    for (int vert = 0; vert < lenArr; ++vert) { NSide3Y[vert] += miny; }
}

void usingTransform(float const minx, float const miny) {
    std::transform(StemX.begin(), StemX.end(), StemX.begin(), [minx](float x){return x + minx;});
    std::transform(StemX.begin(), StemX.end(), StemX.begin(), [miny](float x){return x + miny;});
    std::transform(Head1X.begin(), Head1X.end(), Head1X.begin(), [minx](float x){return x + minx;});
    std::transform(Head1Y.begin(), Head1Y.end(), Head1Y.begin(), [miny](float x){return x + miny;});
    std::transform(Head2X.begin(), Head2X.end(), Head2X.begin(), [minx](float x){return x + minx;});
    std::transform(Head2Y.begin(), Head2Y.end(), Head2Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide1X.begin(), NSide1X.end(), NSide1X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide1Y.begin(), NSide1Y.end(), NSide1Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide2X.begin(), NSide2X.end(), NSide2X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide2Y.begin(), NSide2Y.end(), NSide2Y.begin(), [miny](float x){return x + miny;});
    std::transform(NSide3X.begin(), NSide3X.end(), NSide3X.begin(), [minx](float x){return x + minx;});
    std::transform(NSide3Y.begin(), NSide3Y.end(), NSide3Y.begin(), [miny](float x){return x + miny;});
}

int main() {
    constexpr unsigned int timesToRepeat = 1000000;
    std::chrono::time_point<std::chrono::system_clock> start, end;
    std::cout << "Starting test!" << std::endl;

    std::cout << "Aggregating Arrays 1... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        multipleVectorsAtOnce(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration2 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration2 << "ms" << std::endl;

    std::cout << "Aggregating Arrays 2... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        vectorsSeparately(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration3 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration3 << "ms" << std::endl;

    std::cout << "Aggregating Arrays 3... ";
    start = std::chrono::system_clock::now();
    for (int i = 0; i < timesToRepeat; i++) {
        usingTransform(1.0, 2.0);
    }
    end = std::chrono::system_clock::now();
    auto duration4 = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
    std::cout << "Done in " << duration4 << "ms" << std::endl;

    return 0;
}
```

Looks like at this point, everything flattens out and it doesn't really matter which way we go:

```
Starting test!
Aggregating Arrays 1... Done in 8ms
Aggregating Arrays 2... Done in 9ms
Aggregating Arrays 3... Done in 8ms
```